packages/expo-doctor/src/checks/IllegalPackageCheck.ts

```diff
@@ -0,0 +1,37 @@
+import { warnAboutDeepDependenciesAsync } from '../utils/explainDependencies';
+import { gteSdkVersion } from '../utils/gteSdkVersion';
+import { DoctorCheck, DoctorCheckParams, DoctorCheckResult } from './checks.types';
+
+export class IllegalPackageCheck implements DoctorCheck {
+  description = 'Checking for incompatible packages';
+
+  async runAsync({ exp, projectRoot }: DoctorCheckParams): Promise<DoctorCheckResult> {
+    const issues: string[] = [];
+    const advice: string[] = [];
+
+    if (gteSdkVersion(exp, '44.0.0')) {
+      const illegalPackages = [
+        '@unimodules/core',
+        '@unimodules/react-native-adapter',
+        'react-native-unimodules',
+      ];
+
+      for (const pkg of illegalPackages) {
+        const warning = await warnAboutDeepDependenciesAsync({ name: pkg }, projectRoot);
+        if (warning) {
+          issues.push(warning);
```

Comment:
I don't understand this logic. `warnAboutDeepDependenciesAsync` returns the result of `printExplanationAsync` (which is either null or a string of concatenated issue lines). That is being pushed into an array here, which is then only used to check its length later. That means this whole issues array is actually just a flag 'issuesFound' that can get set in the loop.